paramiko/file.py

```diff
@@ -372,7 +372,10 @@ def write(self, data):
 
         :param str/bytes data: data to write
         """
-        data = b(data)
+        if isinstance(data, text_type):
+            # GZ 2017-05-25: Accepting text on a binary stream unconditionally
+            # cooercing to utf-8 seems questionable, but compatibility reasons?
```

Comment:
Well, on Python 2, it well might be `str`, thus the confusion :D but yea, not really that worried, just making a note of it.